src/angularD3/directives/chart.ts

```diff
@@ -33,6 +33,7 @@ export class D3Chart implements OnInit {
     this.element = elementRef.nativeElement;
     this.chart = d3.select(this.element).attr('class', "d3-chart")
     window.addEventListener('resize', () => this.redraw())
+    window.addEventListener('a2d3RedrawRequest', () => this.redraw())
```

Comment:
Is there no Angular 2 native way to do this without adding a global event? Can the owner of the chart not trigger this directly using the reference to the component in the page?